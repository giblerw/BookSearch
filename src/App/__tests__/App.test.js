import { shallowMount, mount } from '@vue/test-utils';
import AppComponent from '../index.vue';

describe('App.vue', () => {
    it('renders the app title text correctly', () => {
        const wrapper = shallowMount(AppComponent);
        expect(wrapper.find('[id="app-title"]').text())
        .toEqual('Bookz');
    });

    it('renders the HTML structure correctly', () => {
        const wrapper = mount(AppComponent);
        expect(wrapper.element).toMatchSnapshot();
    })
});