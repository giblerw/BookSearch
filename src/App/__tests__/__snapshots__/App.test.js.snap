// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders the HTML structure correctly 1`] = `
<div>
  <header
    class="row"
    id="app-header"
  >
    <h3
      class="col-lg-3"
      id="app-title"
    >
      Bookz
    </h3>
     
    <div
      class="search-bar-container col-lg-9"
      id="search-bar"
    >
      <form>
        <input
          placeholder="Search here"
          type="search"
        />
         
        <i
          class="fa fa-search"
        >
          <button
            id="search-submit"
            type="submit"
          />
        </i>
      </form>
       
      <!---->
    </div>
  </header>
   
  <div
    id="list-container"
  >
    <div
      class="list-prompt"
    >
      <p
        id="search-propmt"
      >
        
    Submit a search above to get Book data.
  
      </p>
       
      <!---->
    </div>
  </div>
</div>
`;
